diff --git a/lib/index.mjs b/lib/index.mjs
index 1857136..90777a3 100644
--- a/lib/index.mjs
+++ b/lib/index.mjs
@@ -6,6 +6,6 @@ import Renderer from './renderer.mjs'
 import ParserCore from './parser_core.mjs'
 import ParserBlock from './parser_block.mjs'
 import ParserInline from './parser_inline.mjs'
-import LinkifyIt from 'linkify-it'
+import { LinkifyIt } from 'linkify-it'
 import * as mdurl from 'mdurl'
 import punycode from 'punycode.js'
diff --git a/lib/rules_core/linkify.mjs b/lib/rules_core/linkify.mjs
index 052c3d8..9873626 100644
--- a/lib/rules_core/linkify.mjs
+++ b/lib/rules_core/linkify.mjs
@@ -20,5 +20,5 @@
   for (let j = 0, l = blockTokens.length; j < l; j++) {
     if (blockTokens[j].type !== 'inline' ||
-        !state.md.linkify.pretest(blockTokens[j].content)) {
+        !state.md.linkify.test(blockTokens[j].content)) {
       continue
     }
