diff --git a/src/constants.js b/src/constants.js
index 2ea4168a931831a31fef582de627af82ea8ef95b..57d5c2e116c35cbb09b6e4b0943e951ad4bdcc67 100644
--- a/src/constants.js
+++ b/src/constants.js
@@ -13,7 +13,7 @@ export const AUTH_HEADER_KEY = "Authorization";
 export const CSRF_KEY = "csrftoken";
 export const DEFAULT_RECORD_LIMIT = 100;
 export const ERRORS_KEY = "error_code";
-export const BASE_URL = "https://api.monarchmoney.com"
+export const BASE_URL = "https://api.monarch.com"
 export const GQL_ENDPOINT = `${BASE_URL}/graphql`;
 
 export const MonarchMoneyEndpoints = {
